### PR TITLE
Move loop into CommandHandler

### PR DESCRIPTION
**Pull request by kalikiana**
See: https://progress.opensuse.org/issues/81899

Note: Requires #2196

### CODE REVIEW

File: isotovideo [hunk 1]
```diff
@@ -262,92 +246,33 @@ $backend = OpenQA::Isotovideo::Backend->new;
...
-    $rsp->{tags} = $command_handler->tags;
-    if ($rsp->{found} || $rsp->{timeout}) {
-        myjsonrpc::send_json($testfd, {ret => $rsp});
-        $command_handler->clear_tags_and_timeout();
-    }
-    else {
-        _calc_check_delta unless $no_wait;
-    }
-}
+$command_handler->on(signal => sub ($event, $sig) {
```
Comment (Martchus) [inline isotovideo:266]: The event you're emitting is just called `sig` so this handler won't be executed.

**Review by Martchus — COMMENTED** [review]
The use of the events looks generally good (especially considering events are already used here anyways).

File: isotovideo (hunk 1 above)
Comment (Martchus) [reply isotovideo:266]: It looks still inconsistent. The signal is now called `signalhandler`.

File: OpenQA/Isotovideo/CommandHandler.pm [hunk 2]
```diff
@@ -53,6 +58,28 @@ has backend_requester => undef;
...
+    my $self = $class->SUPER::new(@args);
+
+    my $signalhandler = sub ($sig) { $self->_signalhandler($sig) };
+    $SIG{TERM} = $signalhandler;
+    $SIG{INT} = $signalhandler;
+    $SIG{HUP} = $signalhandler;
+
+    my ($last_check_seconds, $last_check_microseconds) = gettimeofday;
+    $self->last_check_seconds($last_check_seconds);
+    $self->last_check_microseconds($last_check_microseconds);
```
Comment (Martchus) [inline OpenQA/Isotovideo/CommandHandler.pm:74]: An explicit return would be nice.

File: OpenQA/Isotovideo/CommandHandler.pm [hunk 3]
```diff
@@ -53,6 +58,28 @@ has backend_requester => undef;
...
+# the time of the last asserted screen
+has [qw(last_check_seconds last_check_microseconds)] => 0;
+
+sub new ($class, @args) {
+    my $self = $class->SUPER::new(@args);
+
+    my $signalhandler = sub ($sig) { $self->_signalhandler($sig) };
+    $SIG{TERM} = $signalhandler;
+    $SIG{INT} = $signalhandler;
+    $SIG{HUP} = $signalhandler;
```
Comment (Martchus) [inline OpenQA/Isotovideo/CommandHandler.pm:70]: I'd extract that into a function, e.g. called `setup_signal_handler` and have the user calling it after constructing the object. (Assigning global variables unconditionally is not what I'd generally expect when invoking a constructor.)

**Review by Martchus — CHANGES_REQUESTED** [review]

File: OpenQA/Isotovideo/CommandHandler.pm (hunk 2 above)
Comment (kalikiana) [reply OpenQA/Isotovideo/CommandHandler.pm:74]: That's code from #2196 :-)

**Review by Martchus — APPROVED** [review]

**Review by okurz — CHANGES_REQUESTED** [review]
codecov/project/fully_covered failed

**Comment by okurz** [comment]
Now patch coverage :)

**Comment by kalikiana** [comment]
> Now patch coverage :)

Well, if you consider it useful I'll mark it as uncoverable. I don't fully understand why it shows up even though we isotovideo is already exempt...

**Review by okurz — APPROVED** [review]

**Pull request merged**